source/docs/codeberg-pages.md

```diff
@@ -0,0 +1,177 @@
+---
+title: Codeberg Pages
+---
+Push your website to your Codeberg repository at `ssh://git@codeberg.org/username/repository.git`.
+
+By default, Codeberg Pages will serve the `main` branch of `pages` repository under a username or organization at `https://username.codeberg.page` . Any content at any other branch or repository will be served under `https://username.codeberg.page/repository/@branch` . You might noticed that it had been possible to visit `https://branch.repository.username.codeberg.page` but this has been deprecated since `codeberg.page` implemented HSTS, and their wildcard certificate is not covering 4th or 5th level sub-domain.
+
+If you do not need a custom domain and is willing to make use of `username.codeberg.page` subdomain, please consider using a branch other than "main" to host your website root directory, and make sure `public` folder will be using "main" branch.
+
+To use Codeberg Pages to deploy your website with a custom domain, add a `.domains` file at the `/source` folder. If your website is at `pages` branch of `pages` repository, it should contain the following content: 
+
+```text
+example.org
+www.example.org
+username.codeberg.page
+pages.username.codeberg.page
+pages.pages.username.codeberg
+```
+
+In practice, if you only need it to be available at one domain, you can simply add one line only. If you need to redirect `www` subdomain or `username.codeberg.page` to the main domain, please include all the domains for redirection.
+
+And set it as `skip_render`:
+
+```yaml
+skip_render:
+ - '.domains'
+```
+
+# Using Forgejo Actions
```

Comment:
Probably an extra paragraph here before the title, mentioning that this can be done through two different CI, and whether they are mutually exclusive or not. Otherwise, it feels like two CI need to be set up simultaneously.